routers/api/v1/explore/code.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package explore
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/models/db"
+	repo_model "code.gitea.io/gitea/models/repo"
+	code_indexer "code.gitea.io/gitea/modules/indexer/code"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/services/context"
+	"code.gitea.io/gitea/services/convert"
+)
+
+// Code explore code
+func Code(ctx *context.APIContext) {
+	// swagger:operation GET /explore/code explore codeSearch
+	// ---
+	// summary: Search for code
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: q
+	//   in: query
+	//   description: keyword
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: fuzzy
+	//   in: query
+	//   description: whether to search fuzzy or strict
+	//   type: boolean
+	// responses:
+	//   "200":
+	//     description: "SearchResults of a successful search"
+	//     schema:
+	//			 "$ref": "#/definitions/ExploreCodeResult"
+	if !setting.Indexer.RepoIndexerEnabled {
+		ctx.NotFound("Indexer not enabled")
+		return
+	}
+
+	keyword := ctx.FormTrim("q")
+
+	isFuzzy := ctx.FormOptionalBool("fuzzy").ValueOrDefault(true)
```

Comment:
We should document the default value of the `fuzzy` query parm in the swagger docs. You can follow what was done in the [repo search](https://github.com/go-gitea/gitea/blob/b270b30aebf42fb45d9a57288cf3dcbfa842a760/routers/api/v1/repo/repo.go#L92) API.